extractor/src/test/java/org/schabi/newpipe/extractor/services/DefaultStreamExtractorTest.java

```diff
@@ -0,0 +1,289 @@
+import static org.schabi.newpipe.extractor.ExtractorAsserts.assertIsValidUrl;
+import static org.schabi.newpipe.extractor.services.DefaultTests.defaultTestListOfItems;
+
+/**
+ * Test for {@link StreamExtractor}
+ */
+public abstract class DefaultStreamExtractorTest extends DefaultExtractorTest<StreamExtractor>
+        implements BaseStreamExtractorTest {
+
+    public abstract StreamType expectedStreamType();
+    public abstract String expectedUploaderName();
+    public abstract String expectedUploaderUrl();
+    public abstract List<String> expectedDescriptionContains(); // e.g. for full links
+    public abstract long expectedLength();
+    public long expectedTimestamp() { return 0; }; // default: there is no timestamp
+    public abstract long expectedViewCountAtLeast();
+    @Nullable public abstract String expectedUploadDate(); // format: "yyyy-MM-dd HH:mm:ss.SSS"
+    @Nullable public abstract String expectedTextualUploadDate();
+    public abstract long expectedLikeCountAtLeast(); // return -1 if ratings are disabled
+    public abstract long expectedDislikeCountAtLeast(); // return -1 if ratings are disabled
+    public boolean expectedHasRelatedStreams() { return true; } // default: there are related videos
+    public int expectedAgeLimit() { return StreamExtractor.NO_AGE_LIMIT; } // default: no limit
+    @Nullable public String expectedErrorMessage() { return null; } // default: no error message
+    public boolean expectedHasVideoStreams() { return true; } // default: there are video streams
+    public boolean expectedHasAudioStreams() { return true; } // default: there are audio streams
+    public boolean expectedHasSubtitles() { return true; } // default: there are subtitles streams
+    public boolean expectedHasFrames() { return true; } // default: there are frames
+    public boolean expectedHasNextStream() { return true; } // default: there is a next video
+
+    @Test
+    @Override
+    public void testStreamType() throws Exception {
+        assertEquals(expectedStreamType(), extractor().getStreamType());
+    }
+
+    @Test
+    @Override
+    public void testUploaderName() throws Exception {
+        assertEquals(expectedUploaderName(), extractor().getUploaderName());
+    }
+
+    @Test
+    @Override
+    public void testUploaderUrl() throws Exception {
+        final String uploaderUrl = extractor().getUploaderUrl();
+        assertIsSecureUrl(uploaderUrl);
+        assertEquals(expectedUploaderUrl(), uploaderUrl);
+    }
+
+    @Test
+    @Override
+    public void testUploaderAvatarUrl() throws Exception {
+        assertIsSecureUrl(extractor().getUploaderAvatarUrl());
+    }
+
+    @Test
+    @Override
+    public void testThumbnailUrl() throws Exception {
+        assertIsSecureUrl(extractor().getThumbnailUrl());
+    }
+
+    @Test
+    @Override
+    public void testDescription() throws Exception {
+        Description description = extractor().getDescription();
```

Comment:
Could be `final`, just like a lot of variables in this class

Reply:
Ok, will fix this in all modified files